code/modules/mob/living/carbon/xenomorph/Embryo.dm

```diff
@@ -170,52 +172,62 @@
 
 	stage = 6
 
-/mob/living/carbon/Xenomorph/Larva/proc/chest_burst(mob/living/carbon/victim)
-	set waitfor = 0
-	if(victim.chestburst || loc != victim) return
+
+/mob/living/carbon/Xenomorph/Larva/proc/initiate_burst(mob/living/carbon/victim)
+	if(victim.chestburst || loc != victim)
+		return
+
 	victim.chestburst = 1
 	to_chat(src, "<span class='danger'>You start bursting out of [victim]'s chest!</span>")
+
 	if(victim.knocked_out < 1)
 		victim.KnockOut(20)
 	victim.visible_message("<span class='danger'>\The [victim] starts shaking uncontrollably!</span>", \
 								 "<span class='danger'>You feel something ripping up your insides!</span>")
 	victim.Jitter(300)
-	sleep(30)
-	if(!victim || !victim.loc) return//host could've been deleted, or we could've been removed from host.
+
+	addtimer(CALLBACK(src, .proc/burst, victim), 3 SECONDS)
+
+
+/mob/living/carbon/Xenomorph/Larva/proc/burst(mob/living/carbon/victim)
+	if(!victim || !victim.loc)
+		return
+
 	if(loc != victim)
 		victim.chestburst = 0
 		return
+
 	victim.update_burst()
-	sleep(6) //Sprite delay
-	if(!victim || !victim.loc) return
-	if(loc != victim)
-		victim.chestburst = 0 //if a doc removes the larva during the sleep(6), we must remove the 'bursting' overlay on the human
-		victim.update_burst()
-		return
 
-	if(isyautja(victim)) victim.emote("roar")
-	else victim.emote("scream")
-	forceMove(get_turf(victim)) //moved to the turf directly so we don't get stuck inside a cryopod or another mob container.
+	if(isyautja(victim))
+		victim.emote("roar")
+	else
+		victim.emote("scream")
+
+	forceMove(get_turf(victim)) //Moved to the turf directly so we don't get stuck inside a cryopod or another mob container.
 	playsound(src, pick('sound/voice/alien_chestburst.ogg','sound/voice/alien_chestburst2.ogg'), 25)
 	round_statistics.total_larva_burst++
 	var/obj/item/alien_embryo/AE = locate() in victim
+
 	if(AE)
 		qdel(AE)
+
 	if(ishuman(victim))
 		var/mob/living/carbon/human/H = victim
 		var/datum/internal_organ/O
 		var/i
-		for(i in list("heart","lungs")) //This removes (and later garbage collects) both organs. No heart means instant death.
+		for(i in list("heart", "lungs")) //This removes (and later garbage collects) both organs. No heart means instant death.
```

Comment:
`for(var/i in list("heart", "lungs"))`, and remove the above definition, since it's unused outside this loop.